### ISSUE TITLE
feat: Smoothed early stopping (EMA of validation metric)

### ISSUE THREAD

**Issue by dndungu** [zerfoo/zerfoo#47]
## Problem

Current early stopping uses raw val_corr, which oscillates wildly across epochs. A single bad epoch can trigger premature stopping, while a single lucky epoch can delay stopping when the model is actually degrading. This makes early stopping unreliable for noisy financial data.

## Proposed Solution

Apply EMA smoothing to the validation metric before making early stopping decisions:

```
smoothed_val_corr = es_decay * smoothed_val_corr + (1 - es_decay) * raw_val_corr
```

- Default es_decay: 0.8
- Compare smoothed values for best-model checkpointing
- Use smoothed values for patience countdown
- Log both raw and smoothed values for debugging

## Implementation Details

- Add `es_decay` config field (default 0.8, 0.0 = disabled / use raw values)
- Modify the early stopping logic in the training loop to use smoothed metric
- No changes to the optimizer or model -- this is purely a training loop change
- Should work with any validation metric (val_corr, val_loss, etc.)

## Expected Impact

Quick win (0.5 days). Prevents premature stopping on noisy validation data. Most effective when combined with EMA of weights (#16).

## Context

- Numerai tournament experiments 1-19
- val_corr oscillates +/-0.010 across epochs making raw early stopping unreliable
